include/items.php

```diff
@@ -845,15 +846,26 @@ function item_store($arr,$force_parent = false, $notify = false, $dontcache = fa
 		}
 	} else {
 		// This can happen - for example - if there are locking timeouts.
-		logger("Item wasn't stored - we quit here.");
-		q("COMMIT");
+		q("ROLLBACK");
+
+		// Store the data into a spool file so that we can try again later.
```

Comment:
I'd definitely could assist you in designing a class architecture. I would have to spend a good deal of time studying the current item functions however. So far I only scanned them and changed the parts that were only directly relevant to the issue I was interested in. I'll try to come up with a little design document to make sure I'm not forgetting anything.